clinica/utils/freesurfer.py

```diff
@@ -218,289 +284,169 @@ def extract_image_id_from_longitudinal_segmentation(freesurfer_id):
-            'meanCurv'. Not used for now. Might be used as part of a
-            pandas data frame
-        scalar_list (list of float): list of values corresponding to the keys
+        .. note::
+            This will be an empty string if the line does not contain
+            specific keywords related to the type of information.
+
+    value : str
+        The corresponding statistics value.
+
+        .. note::
+            This will be an empty string if the line does not contain
+            specific keywords related to the type of information.
     """
-    import warnings
+    info_to_keywords = {
+        InfoType.VOLUME: ["volume", "Volume"],
+        InfoType.AREA: ["area", "Area"],
+        InfoType.THICKNESS: ["thickness", "Thickness"],
+    }
+    word_list = line.replace(",", "").split()
+    if any(x in word_list for x in info_to_keywords[info]):
+        return word_list[2], word_list[-2]
+    return "", ""
+
+
+def get_secondary_stats(stats_filename: PurePath, info_type: InfoType) -> pd.DataFrame:
+    """Read the 'secondary' statistical info from .stats file.
+
+    Extract the information from .stats file that is commented out
+    (lines starting with '# Measure' prefix) and does not appear in the
+    table at the end of the document.
+
+    Parameters
+    ----------
+    stats_filename : PurePath
+        Path to the .stats file.
 
-    import pandas
+    info_type : InfoType
+        Type of information to read from .stats file.
 
-    try:
-        data = pandas.DataFrame({"label_name": name_list, "label_value": scalar_list})
-        data.to_csv(out_filename, sep="\t", index=False, encoding="utf-8")
-    except Exception as exception:
-        warnings.warn("Impossible to save {0} file".format(out_filename))
-        raise exception
+    Returns
+    -------
+    secondary_stats_dict : pd.DataFrame
+        DataFrame containing the regions of the brain (in a column named 'label_name'),
+        and the corresponding volume/thickness/area values depending on the
+        input info type (in a column named 'label_value').
+    """
+    if info_type == InfoType.MEANCURV:
+        return pd.DataFrame()
 
-    return out_filename
+    with open(stats_filename, "r") as stats_file:
+        stats = stats_file.read()
+    stats_lines = [
+        line for line in stats.splitlines() if _stats_line_filter(line, info_type)
+    ]
+    regions_stats = [
+        _extract_region_and_stat_value(line, info_type) for line in stats_lines
+    ]
+    regions = [r[0] for r in regions_stats if r[0] != ""]
+    values = [r[1] for r in regions_stats if r[1] != ""]
+    if len(regions) != len(values):
+        raise ValueError(
+            f"In stats file {stats_filename}, the extracted secondary stats "
+            f"have different lengths {len(regions)} vs. {len(values)}. "
+            "Please check the lines starting with '# Measure' in this file."
+        )
+    return pd.DataFrame({"label_name": regions, "label_value": values})
```

Comment:
The whole thing could be simplified to this:

```suggestion
    return pd.DataFrame.from_records(regions_stats, columns={"label_name", "label_value"})
```

This logic looks very unsafe to me, as we can literally shift one column with respect to the other accidentally.

```python
    regions = [r[0] for r in regions_stats if r[0] != ""]
    values = [r[1] for r in regions_stats if r[1] != ""]
``` 

In my opinion, `_extract_region_and_stat_value` should be responsible for filtering lines containing empty values and return proper records.

Reply:
I suspect that `_extract_region_and_stat_value` can only return `("", "")` and not `("", "right_val")` or `("left_val", "")`. But the code should prevent invalid states be computed in the first place, hence why splitting the columns and filtering them individually is wrong in my opinion.

Instead, `_extract_region_and_stat_value` should return records without empty values, which would require no further splitting or filtering.

Reply:
Thanks for the suggestion @ghisvail ! Using `from_records` is much better indeed ! 👍 
For the second point, I also agree. I made `_extract_region_and_stat_value` return `None` if nothing could be extracted. The list of tuples then need to be cleaned from `Nones` to be a valid input to the data frame constructor. Hope this answers your concerns.